client/lib/accept/dialog.jsx

```diff
@@ -29,11 +29,8 @@ const AcceptDialog = React.createClass( {
 	},
 
 	onClose: function( action ) {
+		this.setState( { isVisible: false } );
 		this.props.onClose( 'accept' === action );
```

Comment:
`onClose` is [provided by](https://github.com/Automattic/wp-calypso/blob/60ea236ee15be699b2e7a20fb668e5f920208dc5/client/lib/accept/index.js#L19-L29) `lib/accept/index.js` and unmounts this component, so it shouldn't come as a surprise that after this line has run, the component is no longer mounted. Moving `setState` before unmounting should fix this.